lib/src/glossary/widgets/search_bar.dart

```diff
@@ -0,0 +1,279 @@
+    if (!isFocusOn) {
+      newWidth = newWidth - 94.0;
+    }
+    if (isSearchMade) {
+      newWidth = newWidth - 40;
+    }
+    return newWidth;
+  }
+
+  void changeGlass() {
+    if (!isOpen) {
+      isOpen = true;
+      _focusNode.requestFocus();
+      isFocusOn = true;
+    } else if (_controller.text.isEmpty) {
+      isOpen = false;
+      _focusNode.unfocus();
+      isFocusOn = false;
+      widget.searchGlossary("");
+    } else {
+      isSearchMade = true;
+      _focusNode.unfocus();
+      isFocusOn = false;
+    }
+    changeAnimationState();
+  }
+
+  void clearSearch() {
+    _controller.clear();
+    isInputFilled = false;
+    isSearchMade = false;
+    _focusNode.unfocus();
+    _focusNode.requestFocus();
+    focusInSearch();
+  }
+
+  void focusOutSearch(event) {
+    _focusNode.unfocus();
+    isFocusOn = false;
+    changeAnimationState();
+  }
+
+  void focusInSearch() {
+    isFocusOn = true;
+    changeAnimationState();
+  }
+
+  void cancelSearch() {
+    isSearchMade = false;
+    clearSearch();
+    focusOutSearch(null);
+    isOpen = false;
+    changeAnimationState();
+    widget.searchGlossary("");
+  }
+
+  void onSubmit(String searchTxt) {
+    if (searchTxt != "") {
+      isSearchMade = true;
+      focusOutSearch(null);
+      widget.searchGlossary(searchTxt);
+    } else {
+      _focusNode.requestFocus();
+    }
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    _focusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations l10n = AppLocalizations.of(context);
+
+    return Stack(
+      children: [
+        SizedBox(
+          height: 64,
+          child: Row(
+            children: [
+              Expanded(
+                flex: 1,
+                child: Container(),
+              ),
+              Expanded(
+                flex: 2,
+                child: Center(
+                  child: Text(l10n.glossary_view_title),
+                ),
+              ),
+              Expanded(
+                  flex: 1,
+                  child: Row(
+                    mainAxisAlignment: MainAxisAlignment.end,
+                    children: [
+                      IconButton(
+                        onPressed: () => {},
+                        icon: const Icon(Icons.filter_list_rounded),
+                      ),
+                      IconButton(
+                        onPressed: () => {},
+                        icon: const Icon(Icons.tune),
+                      ),
+                    ],
+                  )),
+            ],
+          ),
+        ),
+        if (isOpen && isSearchMade)
+          InkWell(
+            key: const Key("glossary_bar_cancel_search_icon"),
+            hoverColor: Colors.transparent,
+            splashColor: Colors.transparent,
+            focusColor: Colors.transparent,
+            highlightColor: Colors.transparent,
+            onTap: () => cancelSearch(),
```

Comment:
Pourquoi mettre une lambda et pas? 

```suggestion
            onTap: cancelSearch,
```

Reply:
Bonne question, c'est changé